Integration of DESEQ2 module

tborrman-epic opened the pull request:
Integration of DESEQ2 module has been tested and completed successfully with the following test data run:

```bash
nextflow run https://github.com/epicrispr-biotechnologies/epic-rnaseq \
    -r deseq2 \
    --user tborrman \
    --study JIRA_XXXX_study \
    --input s3://epic-bio-data-us-west-2/nextflow/nextflow_data/off-target-pipeline/RNAseq_1M_samplesheet.csv \
    --outdir s3://epic-bio-data-us-west-2/tborrman/test_epic-rnaseq_deseq2_RNAsesq_1M/ \
    -bucket-dir s3://epic-bio-data-us-west-2/tborrman/test/ \
    --fasta s3://epic-bio-data-us-west-2/ref_genome_index/human/ensembl/release_109/genome_fasta/Homo_sapiens.GRCh38.dna.primary_assembly.109.cas_molecule.fa \
    --gtf s3://epic-bio-data-us-west-2/ref_genome_index/human/ensembl/release_109/gene_annotation/Homo_sapiens.GRCh38.109.cas_molecule.gtf \
    --star_index s3://epic-bio-data-us-west-2/ref_genome_index/human/ensembl/release_109/indices/star_index \
    --salmon_index s3://epic-bio-data-us-west-2/ref_genome_index/human/ensembl/release_109/indices/salmon_index \
    -profile awsbatch,docker \
    -latest
```

### Quick summary of changes:

- New module DESEQ2_REPORT
- New docker image to run DESEQ2_REPORT added to ECR
- Updated README documentation
- New input params required for running DESEQ2_REPORT
- New input samplesheet style format

### WARNINGS
- Only tested using `awsbatch` and `docker` profiles, not tested for `conda` and no support for `singularity` 
- No unit testing

tborrman-epic committed:
initial deseq2 markdown code

tborrman-epic committed:
working knit for 1ABIC with EPI-321 cargo

tborrman-epic committed:
added pre filtering back and cleaned code

tborrman-epic committed:
changing alpha=0.001 to alpha=0.01

tborrman-epic committed:
aws support test 1

tborrman-epic committed:
aws support #2

tborrman-epic committed:
fixing s3 workdir and making it local

tborrman-epic committed:
remove workDir in config, using -bucket-dir arg in command line

tborrman-epic committed:
pretty markdown

tborrman-epic committed:
working pretty

tborrman-epic committed:
no more hard coded paths

tborrman-epic committed:
increasing time limit for STAR_ALIGN process

tborrman-epic committed:
full path for updating star align

tborrman-epic committed:
add deseq2 table

tborrman-epic committed:
scree plot and cleaned up figure descriptions

tborrman-epic committed:
adjusting fig sizes

tborrman-epic committed:
explicit contrasts declarations fixed

tborrman-epic committed:
adding top 25 genes and cleaned up package installations

tborrman-epic committed:
24.h

tborrman-epic committed:
custom config with 24.h

tborrman-epic committed:
changing test.config

tborrman-epic committed:
gunzip gtf test

tborrman-epic committed:
config 1

tborrman-epic committed:
config 2

tborrman-epic committed:
config 2

tborrman-epic committed:
config 3

tborrman-epic committed:
config 3

tborrman-epic committed:
config 4

tborrman-epic committed:
config 5

tborrman-epic committed:
config 6

tborrman-epic committed:
config 7

tborrman-epic committed:
working config params

tborrman-epic committed:
rm custom config

tborrman-epic committed:
Merge pull request #1 from SuperSecretBioTech/test_config

Test config

tborrman-epic committed:
new counts matrix datatable and connectors for volcano plot

tborrman-epic committed:
fixing PCA labels, and adding UD/D2 contrast

tborrman-epic committed:
new contrasts declared and edited headers

tborrman-epic committed:
header fix

tborrman-epic committed:
environments

tborrman-epic committed:
no dev volcano

tborrman-epic committed:
deseq2 hello world attempt1

tborrman-epic committed:
deseq2 hello world attempt2

tborrman-epic committed:
deseq2 hello world success

tborrman-epic committed:
working ls star_salmmon

tborrman-epic committed:
deseq2 test 1M

tborrman-epic committed:
docker working, markdown attempt

tborrman-epic committed:
finally first successful pipeline completion

tborrman-epic committed:
user,study, params, clean up

tborrman-epic committed:
container id typo

tborrman-epic committed:
Add files via upload

tborrman-epic committed:
Add files via upload

tborrman-epic committed:
Update README.md

spencer-lopp commented on README.md:
```diff
@@ -78,15 +79,23 @@ Each row represents a fastq file (single-end) or a pair of fastq files (paired e
+nextflow run https://github.com/epicrispr-biotechnologies/epic-rnaseq \
+    --user tborrman \
+    --study JIRA_XXXX_study \
+    --input s3://epic-bio-data-us-west-2/nextflow/nextflow_data/off-target-pipeline/RNAseq_1M_samplesheet.csv \
```
If it's all the same, I'd rather keep the parameters generic and put these specific paths in some kind of example file

tborrman-epic replied:
Fixed!

spencer-lopp reviewed (commented):
looks good, just have the one commend regarding param neatness convention

tborrman-epic committed:
add DEG count

tborrman-epic committed:
labels on pca plot, header text changes

tborrman-epic committed:
Update README.md for generic arguments

zaki-bds committed:
Adding primary cell contrasts

Adding the following contrasts for primary cells, and adding documentation to markdown for allowed condition values:

    Control:Treatment
    UNTREATED:EPI321
    STOP:EPI321
    UNTREATED:STOP

tborrman-epic committed:
thresholds as args with defaults

tborrman-epic committed:
add args to pipeline

tborrman-epic committed:
Update README.md with threshold args
